src/lib.rs

```diff
@@ -225,8 +225,25 @@ impl<T> Slab<T> {
     /// # use slab::*;
     /// let slab: Slab<i32> = Slab::new();
     /// ```
-    pub fn new() -> Slab<T> {
-        Slab::with_capacity(0)
+    #[cfg(not(slab_no_const_vec_new))]
+    pub const fn new() -> Self {
```

Comment:
You could use the [`const-fn`](https://docs.rs/const_fn/latest/const_fn/) crate here to avoid code deduplication.

Reply:
It is my crate, but it doesn't yet implement the behavior I want.

https://github.com/tokio-rs/slab/blob/56e794722ade3f07ecd928d7f66933290e36ac1d/build.rs#L15-L17